webclient/dokument/views.py

```diff
@@ -1073,33 +1084,41 @@ def post(self, request, *args, **kwargs):
         cast.create_transaction(request.user, self.success_message)
         dokument = cast.dokument
         lokalita_update = None
-        if (
-            isinstance(cast.archeologicky_zaznam, ArcheologickyZaznam)
-            and cast.archeologicky_zaznam.typ_zaznamu == ArcheologickyZaznam.TYP_ZAZNAMU_LOKALITA
-            and cast.archeologicky_zaznam.stav == AZ_STAV_ARCHIVOVANY
-            and cast.archeologicky_zaznam.lokalita.igsn
-        ):
-            lokalita_update = cast.archeologicky_zaznam.lokalita
-        if cast.komponenty:
-            komps = cast.komponenty
-            cast.komponenty = None
-            cast.save()
-            komps.delete()
         try:
-            if cast.neident_akce:
-                neident_akce = cast.neident_akce
-                neident_akce: NeidentAkce
-                neident_akce.suppress_signal = True
-                neident_akce.delete()
-        except ObjectDoesNotExist:
-            logger.debug(
-                "dokument.views.DokumentCastSmazatView.post.neident_akce_not_exists",
-                extra={"ident_cely": cast.ident_cely},
+            if (
+                isinstance(cast.archeologicky_zaznam, ArcheologickyZaznam)
+                and cast.archeologicky_zaznam.typ_zaznamu == ArcheologickyZaznam.TYP_ZAZNAMU_LOKALITA
+                and cast.archeologicky_zaznam.stav == AZ_STAV_ARCHIVOVANY
+                and cast.archeologicky_zaznam.lokalita.igsn
+            ):
+                lokalita_update = cast.archeologicky_zaznam.lokalita
+            if cast.komponenty:
+                komps = cast.komponenty
+                cast.komponenty = None
+                cast.save()
+                komps.delete()
+            try:
+                if cast.neident_akce:
+                    neident_akce = cast.neident_akce
+                    neident_akce: NeidentAkce
+                    neident_akce.suppress_signal = True
+                    neident_akce.delete()
+            except ObjectDoesNotExist:
+                logger.debug(
+                    "dokument.views.DokumentCastSmazatView.post.neident_akce_not_exists",
+                    extra={"ident_cely": cast.ident_cely},
+                )
+            cast.close_active_transaction_when_finished = True
+            cast.delete()
+            if lokalita_update:
```

Comment:
Nemělo by tohle být až za těmi exception, aby se to spustilo jen tehdy, pokud vše projde?

Reply:
Ale pokud selže úprava IGSN, tak zase z pohledu aplikace bude vše v pořádku a přitom IGSN nebude aktualizované.